test/int/cluster_test.go

```diff
@@ -476,6 +478,42 @@ var _ = Describe("AtlasCluster", func() {
 		})
 	})
 
+	Describe("Create DBUser before cluster & check secrets", func() {
+		It("Should Succeed", func() {
+			By(fmt.Sprintf("Creating password Secret %s", UserPasswordSecret), func() {
+				passwordSecret := buildPasswordSecret(UserPasswordSecret, DBUserPassword)
+				Expect(k8sClient.Create(context.Background(), &passwordSecret)).To(Succeed())
+			})
+
+			createdDBUser := mdbv1.DefaultDBUser(namespace.Name, "test-db-user", createdProject.Name).WithPasswordSecret(UserPasswordSecret)
```

Comment:
can you please extend the test case and create two users - one with empty scope and another - with non-empty and referencing the fake cluster. The later one will fail with validation which is ok. We need to make sure that the connection secret for it wasn't built.